pkg/core/transaction/witness.go

```diff
@@ -38,3 +40,11 @@ func (w *Witness) EncodeBinary(bw *io.BinWriter) {
 func (w Witness) ScriptHash() util.Uint160 {
 	return hash.Hash160(w.VerificationScript)
 }
+
+// Copy creates a deep copy of the Witness.
+func (w *Witness) Copy() *Witness {
```

Comment:
We need to define it on structure instead of a pointer, and return a structure instead of pointer so that the result can be directly used to initialize new tx's witness without dereference.